scripts/BooksAPI.php

```diff
@@ -35,14 +35,14 @@ function getBooks()
     $url = $this->createRequestURL();
     $array = $this->get($url);
     $nrItems = $array['totalItems'];
-    do { //< $nrItems
+    while ($this->startIndex === NULL || $this->startIndex < $nrItems) {
```

Comment:
$nrItems should be checked if it's NULL and for that you have is_null() function.